### PR TITLE
Swagger spec change for Namespace Sku and patch op

### PR DESCRIPTION
**Pull request by amolr**
Includes the newly incorporated Sku change in the namespace and patch operation for namespaces

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@amolr__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (amolr). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 1]
```diff
@@ -129,6 +129,52 @@
           }
         }
       },
+	  "patch": {
```
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: indentation 2 spaces.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 2]
```diff
@@ -129,6 +129,52 @@
...
+            "description": "Parameters supplied to create a Namespace Resource."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+		"responses": {
```
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: replace tabs with spaces and indentation 2 spaces

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 3]
```diff
@@ -129,6 +129,52 @@
...
+			"200": {
+				"description": "Namespace is Create/Update",
```
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: "Namespace is updated"

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 4]
```diff
@@ -1791,7 +1855,32 @@
...
+		"type":"string",
+		"description": "Name of the notification hub sku"
+	 },
+	 "tier": {
+		"type":"string",
+		"description":"Name of the notification hub tier"
+	 },
+	 "size": {
+		"type":"string",
+		"description":""
```
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: general comment: empty descriptions are not allowed.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 5]
```diff
@@ -1791,7 +1855,32 @@
         }
       },
       "x-ms-azure-resource": true
-    }
+    },
+	"Sku": {
+	 "properties": {
```
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: What are the required properties of "Sku"?

Can name, tier, family, capacity, size be enums? Do they have a finite set of values? How will the user know what values to provide?

**Review by amarzavery — CHANGES_REQUESTED** [review]

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 6]
```diff
@@ -129,6 +129,52 @@
           }
         }
       },
+	  "patch": {
+		"tags": [
+          "Namespaces"
+        ],
+		"operationId": "Namespaces_Patch",
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Namespace_Update, Patch seems too technical

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 7]
```diff
@@ -129,6 +129,52 @@
...
+		"description": "Patches the existing namespace",
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Updates

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 8]
```diff
@@ -129,6 +129,52 @@
...
+		"parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: @amarzavery Yesterday you were showing me to put something here, is this applicable to this PR as well.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 9]
```diff
@@ -129,6 +129,52 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The namespace name."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NamespacePatchParameters"
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: NamespaceUpdateParameters

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 10]
```diff
@@ -129,6 +129,52 @@
...
+            },
+            "description": "Parameters supplied to create a Namespace Resource."
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Properties to be updated in Namespace resource

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 11]
```diff
@@ -1315,6 +1361,21 @@
...
+          "$ref": "#/definitions/NamespaceProperties",
+          "description": "Properties of the Namespace."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Parameters supplied to the Patch Namespace operation."
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Parameters supplied to the update Namespace resource.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 12]
```diff
@@ -1757,7 +1818,6 @@
           "description": "Resource Id"
         },
         "name": {
-          "readOnly": true,
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Not sure why would you remove it, name from the body does not suppose to be used by RP, so this suppose to be read only.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 13]
```diff
@@ -1776,7 +1836,11 @@
             "type": "string"
           },
           "description": "Resource tags"
-        }
+        },
+		"sku": {
```
Comment (anudeepsharma) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Fix indentation

**Review by anudeepsharma — APPROVED** [review]
Approved with comments.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json (hunk 6 above)
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: That is the recommended pattern. That makes things clear to the customer that it is a PATCH operation.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json (hunk 8 above)
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: @anudeepsharma This [extension](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-parameter-location) should provide you more info.

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json (hunk 12 above)
Comment (amolr) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: This was caused by remote merge

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 14]
```diff
@@ -1790,6 +1856,42 @@
...
+          "type": "string",
+          "description": "Name of the notification hub sku",
+          "enum": [
+            "Free",
+            "free",
+            "Basic",
+            "basic",
+            "Standard",
+            "standard"
+          ]
```
Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]:
```
"x-ms-enum": {
   "name": "SkuName",
   "modelAsString": true
}
```

**Review by amarzavery — APPROVED** [review]

**Pull request merged**